pinot-spi/src/main/java/org/apache/pinot/spi/utils/TimeUtils.java

```diff
@@ -167,6 +169,44 @@ public static Long convertPeriodToMillis(String timeStr) {
     return millis;
   }
 
+  /**
+   * Returns the millisecond of the period since the @since millisecond
+   * For ex, getMillisFromPeriod("2d",
+   *  Clock.fixed(java.time.Instant.ofEpochMilli(1660054290324L), ZoneId.of("UTC"))) -> 1659881490324
+   * The milliseconds of 2d is 172800000 hence the millisecond of 2d since 1660054290324 is
+   * 1660054290324 - 172800000 = 1659881490324
+   *
+   * @param timeStr string representing the duration
+   * @param since Clock representing the time from which the period needs to be computed
+   * @return the corresponding time converted to milliseconds
+   * @throws IllegalArgumentException if the string does not conform to the expected format
+   */
+  public static long getMillisFromPeriod(String timeStr, Clock since) {
+    long periodToMillis = convertPeriodToMillis(timeStr);
+    return since.millis() - periodToMillis;
```

Comment:
Seems we want to go back a certain period from the passed in time. IMO this method is a little bit confusing, so suggest removing it and make the caller use `convertPeriodToMillis()` instead